File: old_trunk/cost/iona/orbacus/cpp/poa/TestMultipleOrbsClient.cpp
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <Test.h>

#include <assert.h>

#include <fstream>

int
main(int argc, char* argv[], char*[])
{
    int status = EXIT_SUCCESS;
    CORBA::ORB_var orb;
    try
    {
	//
	// Create ORB
	//
	orb = CORBA::ORB_init(argc, argv);
	CORBA::Object_var obj;

	//
	// Get test servers object references
	//
	ifstream in("Test.ref");
	if(in.fail())
	{
	    OB_ERROR(argv[0] << ": cannot read Test.ref");
	}

	char s[1000];
	in >> s;
	obj = orb -> string_to_object(s);
	if(CORBA::is_nil(obj))
	{
	    OB_ERROR(argv[0] << ": cannot read IOR from Test.ref");
	    return EXIT_FAILURE;
	}
	TestServer_var server1 = TestServer::_narrow(obj);
	assert(!CORBA::is_nil(server1));

	in >> s;
	obj = orb -> string_to_object(s);
	if(CORBA::is_nil(obj))
	{
	    OB_ERROR(argv[0] << ": cannot read IOR from Test2.ref");
	    return EXIT_FAILURE;
	}
	TestServer_var server2 = TestServer::_narrow(obj);
	assert(!CORBA::is_nil(server2));

        in.close();

	//
	// Get object references from servers
	//
	TestInfoSeq_var info1 = server1 -> get_info();
	TestInfoSeq_var info2 = server2 -> get_info();

	//
	// Call a few methods on each object
	//
	info1[0].obj -> aMethod();
	info2[0].obj -> aMethod();
	info1[0].obj -> aMethod();
	info2[0].obj -> aMethod();

	//
	// Deactivate servers (and thus orbs)
	//
	server1 -> deactivate();
	server2 -> deactivate();
    }
    catch(const CORBA::Exception& ex)
    {
#ifdef HAVE_EXCEPTION_INSERTERS
	OB_ERROR(ex);
#endif
	status = EXIT_FAILURE;
    }

    if(!CORBA::is_nil(orb))
    {
	try
	{
	    orb -> destroy();
	}
	catch(const CORBA::Exception& ex)
	{
#ifdef HAVE_EXCEPTION_INSERTERS
	    OB_ERROR(ex);
#endif
	    status = EXIT_FAILURE;
	}
    }

    return status;
}

File: cost/iona/orbacus/java/poa/TestMultipleOrbsClient.java
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.poa;

import java.util.*;
import java.io.*;

final public class TestMultipleOrbsClient extends test.common.TestBase
{
    public static void
    main(String args[])
    {
        Properties props = System.getProperties();

        org.omg.CORBA.ORB orb = null;

        try
        {
	    //
	    // Create ORB
	    //
	    orb = org.omg.CORBA.ORB.init(args, props);

	    //
	    // Get test servers object references
	    //
	    BufferedReader in = new BufferedReader(new FileReader("Test.ref"));
	    String ref1 = in.readLine();
	    org.omg.CORBA.Object obj = orb.string_to_object(ref1);
	    if(obj == null)
	    {
	        System.err.println("TestMultipleOrbsClient: " +
		                   " cannot read IOR from Test.ref");
		System.exit(1);
	    }
	    TestServer server1 = TestServerHelper.narrow(obj);

	    String ref2 = in.readLine();
	    obj = orb.string_to_object(ref2);
	    if(obj == null)
	    {
	        System.err.println("TestMultipleOrbsClient: " +
		                   " cannot read IOR from Test.ref");
		System.exit(1);
	    }
	    TestServer server2 = TestServerHelper.narrow(obj);

	    in.close();

	    //
	    // Get object references from servers
	    //
	    TestInfo[] info1 = server1.get_info();
	    TestInfo[] info2 = server2.get_info();

	    //
	    // Call a few methods on each object
	    //
	    info1[0].obj.aMethod();
	    info2[0].obj.aMethod();
	    info1[0].obj.aMethod();
	    info2[0].obj.aMethod();

	    //
	    // Deactivate servers (and thus orbs)
	    //
	    server1.deactivate();
	    server2.deactivate();
        }
	catch(java.io.FileNotFoundException ex)
	{
	    System.err.println("TestMultipleOrbsClient: cannot open Test.ref");
	    System.exit(1);
	}
	catch(java.io.IOException ex)
	{
	    System.err.println("TestMultipleOrbsClient: File input error");
	    System.exit(1);
	}
        catch(org.omg.CORBA.SystemException ex)
        {
	    ex.printStackTrace();
	    System.exit(1);
        }

        if(orb != null)
        {
	    try
	    {
	        orb.destroy();
	    }
            catch(org.omg.CORBA.SystemException ex)
	    {
	        ex.printStackTrace();
	        System.exit(1);
	    }
        }

	System.exit(0);
    }
}
